modules/core/shared/src/main/scala/gem/enum/GnirsDisperserOrder.scala

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+package enum
+
+import cats.syntax.eq._
+import cats.instances.string._
+import gem.util.Enumerated
+
+/**
+ * Enumerated type for GNRIS Disperser Order.
+ * @group Enumerations (Generated)
+ */
+sealed abstract class GnirsDisperserOrder(
+  val tag: String,
+  val shortName: String,
+  val longName: String,
+  val count: Int,
+  val defaultWavelength: gem.math.Wavelength,
+  val minWavelength: gem.math.Wavelength,
+  val maxWavelength: gem.math.Wavelength,
+  val deltaWavelength: gem.math.Wavelength,
+  val band: Option[gem.enum.MagnitudeBand],
+  val cross_dispersed: Boolean
```

Comment:
For other `Boolean` cases, we've made an enumerated type unique to that case. For example, see `MosPreimaging`. I guess that would be ideal, but certainly a hassle.  It's probably fine the way it is.

Reply:
Yes, that cross_dispersed boolean everywhere bothers me too. I'll rework it in another PR where I'll try to include the cross-dispersion constrains.

Reply:
Where does this value enter into our calculations? ITC? AGS? I don't know how big a deal this is.

Reply:
This was a late addition, initially we only had the band.

In addition to ITC, it's also used indirectly in [EdCompInstGNIRS.java](https://github.com/gemini-hlsw/ocs/blob/develop/bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gnirs/EdCompInstGNIRS.java#L403-L404). I guess it has to do something with some label in UI, perhaps?

Reply:
Hm. Not really knowing how this is used I guess I'm ok with it staying as a boolean for now. If we end up in a position where we're having to check it a lot and are in danger of forgetting to check (this is the issue I think) then we might want to have distinct types.

Reply:
The thing with cross dispersed mode (XD) is that several instrument parameters are only available when XD is on/off. In this case, some orders are only available in XD mode other orders only when no XD.

For these orders we could split the table in 2: `disperser_order_xd` and `disperser_order_mirror`. But my intention was to rework the whole XD feature everywhere after getting more expertise with GNIRS, because we have more XD constrains with other parameters and I'm not sure what would be the best way to encode them overall.

I was trying to play this PR conservatively and stick with to how it was written in OCS, before trying something more experimental that deviates more from the original code.

I could file ticket for this XD in order to not forget, or we could put this PR on hold and try to work this XD out before merging.

Reply:
> It seems GNIRS behaves as an almost entirely different instrument when the XD is on.

Ok, it sounds like this needs some further design work before we commit to a config implementation. I'm fine merging this as-is, with the expectation that we will revisit it.

Reply:
If the software is to model the instrument we may want to consider replacing the XD "mode" with the actual mechanism, a prism turret with 4 positions:
1. A mirror which is used for long-slit spectroscopy (no cross-dispersion)
2. The SXD prism which cross-disperses orders by ~8"
3. The LXD prism which cross-disperses orders by ~6"
4. The Wollaston prism

Reply:
I'll add Mirror and the Wollaston prism to a new table called `prism_turret`. I could use that table as the main source of constrains for the rest of parameters on other tables.